src/main.c

```diff
@@ -166,8 +166,31 @@ int main(void) {
 #elif defined(SAMD51)
     WDT->CTRLA.reg = 0;
     while(WDT->SYNCBUSY.reg) {}
+
+    // Enable 2.7V brownout detection. The default fuse value is 1.7
+    // Set brownout detection to ~2.7V. Default from factory is 1.7V,
+    // which is too low for proper operation of external SPI flash chips (they are 2.7-3.6V).
+    // Also without this higher level, the SAMD51 will write zeros to flash intermittently.
+    // Disable while changing level.
+    SUPC->BOD33.bit.ENABLE = 0;
+    SUPC->BOD33.bit.LEVEL = 200;  // 2.7V: 1.5V + LEVEL * 6mV.
+    // Don't reset right now.
+    SUPC->BOD33.bit.ACTION = SUPC_BOD33_ACTION_NONE_Val;
+    SUPC->BOD33.bit.ENABLE = 1; // enable brown-out detection
+
+    while (SUPC->STATUS.bit.BOD33DET) {
+        // Wait for voltage to rise above BOD33 value.
+    }
+    // Wait 100ms for voltage to stabilize.
+    delay(100);
+
```

Comment:
Want to do one more final check of the detector here to make sure the voltage is ok?

Reply:
I was thinking of that; I'd repeat until it was stable. Also, I should do this only on certain reset causes, not always. Right now this small delay makes double click tricker (you have to do it slightly slower than you used to).

Reply:
done

Reply:
Ya, checking reset cause would be good. We're going to need something similar in CircuitPython once we reset on events.

Reply:
new commit  now checks reset cause